app/controllers/images_controller.rb

```
@@ -15,15 +20,19 @@ def create
     @image = Image.new(image_params)
 
     if @image.save
+      flash[:success] = 'You have successfully added an image.'
       redirect_to @image
     else
       render 'new', status: 422
     end
   end
 
-  def tagged
-    @tag = params[:tag]
-    @images = @tag.present? ? Image.tagged_with(@tag).order(created_at: :desc) : nil
+  def destroy
+    @image = Image.find(params[:id])
+    @image.destroy
```

For all of these ActiveRecord methods that are making DB changes (e.g. updating a model, creating a model, deleting a model), it is always best to use the "bang" version of the method (i.e. the version that ends with an exclamation point). In this case, that means changing the code from `@image.destroy` to `@image.destroy!` instead. The reason for this change is that all of the "bang" methods will raise an exception if something goes wrong. The non-bang methods will just return `false` and therefore they are more of a silent failure.

There are scenarios where the non-bang versions are preferred (like in the case of your `create` action where you are checking the return value of `@image.save` and doing one of two different things), but in this case the bang version is preferable.

---

changed it to destroy!